MultiServer.py

```diff
@@ -597,6 +550,7 @@ def set_save(self, savedata: dict):
 
         if "stored_data" in savedata:
             self.stored_data = savedata["stored_data"]
+            self.data_storage = DataStorage(self.stored_data)
```

Comment:
... at first glance it is unclear why we have both here and if they are separate or the same.

So, I 'd go `self.data_storage.data` when you want to access it as a raw dict.